docs/topics/object-declarations.md

```diff
@@ -294,65 +237,337 @@ class MyClass {
+
+class NotificationManager {
+    // The return type is Any. The message property is not accessible.
+    // When the return type is Any, only members of the Any class are accessible.
+    fun getNotification() = object {
+        val message: String = "General notification"
+    }
+
+    // The return type is Notification because the anonymous object implements only one interface
+    // The notifyUser() function is accessible because it is part of the Notification interface
+    // The message property is not accessible because it is not declared in the Notification interface
+    fun getEmailNotification() = object : Notification {
+        override fun notifyUser() {
+            println("Sending email notification")
+        }
+        val message: String = "You've got mail!"
+    }
+
+    // The return type is DetailedNotification. The notifyUser() function and the message property are not accessible
+    // Only members declared in the DetailedNotification interface are accessible
+    fun getDetailedNotification(): DetailedNotification = object : Notification, DetailedNotification {
+        override fun notifyUser() {
+            println("Sending detailed notification")
+        }
+        val message: String = "Detailed message content"
+    }
+}
+//sampleEnd
+fun main() {
+    // This produces no output
+    val notificationManager = NotificationManager()
+
+    // The message property is not accessible here because the return type is Any
+    // This produces no output
+    val notification = notificationManager.getNotification()
+
+    // The notifyUser() is accessible
+    // The message property is not accessible here because the return type is Notification
+    val emailNotification = notificationManager.getEmailNotification()
+    emailNotification.notifyUser()
+    // Sending email notification
+
+    // The notifyUser() function and message property are not accessible here because the return type is DetailedNotification
+    // This produces no output
+    val detailedNotification = notificationManager.getDetailedNotification()
+}
+```
+{kotlin-runnable="true"}
+
+### Access variables from anonymous objects
+
+Code within the body of object expressions can access variables from the enclosing scope:
+
+```kotlin
+import java.awt.event.MouseAdapter
+import java.awt.event.MouseEvent
+
+fun countClicks(window: JComponent) {
+    var clickCount = 0
+    var enterCount = 0
+
+    // MouseAdapter provides default implementations for mouse event functions
+    // Simulates MouseAdapter handling mouse events
+    window.addMouseListener(object : MouseAdapter() {
+        override fun mouseClicked(e: MouseEvent) {
+            clickCount++
+        }
+
+        override fun mouseEntered(e: MouseEvent) {
+            enterCount++
+        }
+    })
+    // The clickCount and enterCount variables are accessible within the object expression
+}
+```
+
+## Semantic difference between object expressions and declarations
```

Comment:
```suggestion
## Behavior difference between object expressions and declarations
```
I think "behavior" is more accurate than "semantic" here, what do you think?

Reply:
I like it, I agree - I changed it